SetFlags.cmake

```diff
@@ -74,10 +74,10 @@ macro(set_flags)
 			)
 		endif()
 
-		set(CMAKE_CXX_FLAGS          "${CMAKE_CXX_FLAGS}          -std=c++11")
-		set(CMAKE_CXX_FLAGS_DEBUG    "${CMAKE_CXX_FLAGS_DEBUG}    -std=c++11")
-		set(CMAKE_CXX_FLAGS_COVERAGE "${CMAKE_CXX_FLAGS_COVERAGE} -std=c++11")
-		set(CMAKE_CXX_FLAGS_RELEASE  "${CMAKE_CXX_FLAGS_RELEASE}  -std=c++11")
+		set(CMAKE_CXX_FLAGS          "${CMAKE_CXX_FLAGS}          -std=c++14")
+		set(CMAKE_CXX_FLAGS_DEBUG    "${CMAKE_CXX_FLAGS_DEBUG}    -std=c++14")
+		set(CMAKE_CXX_FLAGS_COVERAGE "${CMAKE_CXX_FLAGS_COVERAGE} -std=c++14")
+		set(CMAKE_CXX_FLAGS_RELEASE  "${CMAKE_CXX_FLAGS_RELEASE}  -std=c++14")
```

Comment:
> You also haven't updated the visual studio version check to reflect the change to require VS2015.

I will do this if this is agreed.

-----

> visual studio still only has partial support

Update 2 says it has full support plus ratified '17 supports too. (It's great :D)

-----

My opinion is that development should not be limited by perceived availability of compiler toolsets. We have a small developer base that is unencumbered by restrictions which say, a company faces, and therefore should not try to artificially apply those restrictions unto ourselves.

Would those who wish to compile it themselves object to requiring a newer version of their compiler, that is, if they don't already have it?